src/system/RoutesModule/Routing/RouteLoader.php

```diff
@@ -342,19 +342,14 @@ private function prependBundlePrefix(Route $route, AbstractBundle $bundle)
      */
     private function sanitizeController($bundleName, $controllerString)
     {
-        if (strpos($controllerString, '::') === false) {
+        if (0 === preg_match('#^(.*?\\\\Controller\\\\(.+)Controller)::(.+)Action$#', $controllerString, $match)) {
             return $controllerString;
         }
-
-        $action = substr($controllerString, strpos($controllerString, '::') + 2);
-        $func = lcfirst(substr($action, 0, -6));
-
-        $a = strrpos($controllerString, '\\') + 1;
-        $b = strrpos($controllerString, '::');
-        $controller = substr($controllerString, $a, $b - $a);
-        $type = substr($controller, 0, -10);
-
-        return $bundleName . ':' . $type . ':' . $func;
+        
+        $controllerName = $match[2];
+        $actionName = $match[3];       
+  
+        return $bundleName . ':' . $controllerName . ':' . $actionName;
```

Comment:
Well for symfony it should be ok, as this fix is taken from symfony. I do not know how it (lcfirst) will affect legacy. I guess we could add lcfirst here https://github.com/Kaik/core/blob/37c4ce4314f09b953656f15459b15ee9dc1c82c3/src/system/RoutesModule/Routing/RouteLoader.php#L265 so it will be more consistent (instead of ```$func``` lcfirst-ed in one place and ```$type``` in other).
Another thing which can be affected is currently commented out loading routes from DB and #2593 issue.